routes/api/users.js

```diff
@@ -0,0 +1,127 @@
+const express = require('express')
+const Joi = require('joi');
+const router = express.Router()
+const jwt = require('jsonwebtoken')
+const passport = require('passport')
+const User = require('../../schemas/userSchema');
+require('dotenv').config()
+const secret = process.env.SECRET
+const {
+    registerUser,
+    findUser,
+    authenticateUser
+} = require("../../models/userController");
+
+const userJoiSchema = Joi.object({
+    email: Joi.string().email().required(),
+    password: Joi.string()
+        .regex(/[0-9a-zA-Z]*\d[0-9a-zA-Z]*/)
+        .min(4)
+        .required(),
+});
+
+const auth = (req, res, next) => {
+    passport.authenticate('jwt', {session: false}, (err, user) => {
+        if (!user || err) {
```

Comment:
Tu trzeba dodać porównanie user.token z tokenem z req, teraz jak się wylogujesz to nadal idzie sprawdzić /current. + przerzucił bym to do jakiegoś osobnego pliku bo tak docelowo to w contacts też chiał byś tego używać.